ushahidi/platform-client
# Feature/2848 hdx fetch surveys

WIP need https://github.com/ushahidi/platform-client/pull/1136 and https://github.com/ushahidi/platform/pull/2923 to be merged

This pull request makes the following changes:
- Adds the tags/attribute-screen to hdx-exports

Testing checklist:
Assuming HDX-feature-flag is ON
- Go to data-export-settings, click on "assign attributes and tags" button
- [x] All surveys (not targeted) and survey-fields should be visible
- [x] No targeted surveys should be visible
- [x] There should be a select-all button, when clicking on that, all fields should become selected
- [x] There should be checkboxes in front of each survey-field
- [ ] The numbers of fields selected should be the same as the count in the text in the end of the table
- [x] After each field there should be two dropdowns, one for tags and one for attributes
- [ ] Each dropdown should be populated with the tags and attributes fetched from the api
- select one field
- [x] the dropdowns should become active
- Select a tag
- [ ] the tag should appear as an icon in the end of the row according to designs
- [ ] the selected tag should be greyed out in the dropdown
- Select an attribute
- [ ] the attribute should be attached to the icon according to designs <--note, this is just a text-list right now, known bug
- [ ] selected attributes should be greyed out in the dropdown
- Click on the yellow x in the icon <--note, this is just a text-list right now, and there is no yellow x, known bug
- [ ] The whole tag should be removed and the dropdowns should contain no greyed out<--note, this is just a text-list right now so cannot be tested, known bug
 tags/attributes
- Select another field
- Select a tag and an attribute
- Click on the "Add another hdx attribute link"
- [ ] Another dropdown should be visible with all attributes available, the previously selected attributes shoud be disabled/greyed out.
- deselect the last attribute
- [ ] this attribute should disappear from the icon
- add another attribute
- [ ] this new attribute should appear in the icon
- select another attribute in the same dropdown
- [ ] the newly selected attribute should appear in the icon instead of the old one

Fixes ushahidi/platform# .

Ping @ushahidi/platform
